tsdb/engine/tsm1/iterator.gen.go

```diff
@@ -422,6 +422,334 @@ func (c *floatLiteralCursor) peek() (t int64, v interface{}) { return tsdb.EOF,
 func (c *floatLiteralCursor) next() (t int64, v interface{}) { return tsdb.EOF, c.value }
 func (c *floatLiteralCursor) nextAt(seek int64) interface{}  { return c.value }
 
+type integerIterator struct {
+	cur   integerCursor
+	aux   []cursorAt
+	conds struct {
+		names []string
+		curs  []*bufCursor
+	}
+	opt influxql.IteratorOptions
+
+	m     map[string]interface{} // map used for condition evaluation
+	point influxql.IntegerPoint  // reusable buffer
+}
+
+func newIntegerIterator(name string, tags influxql.Tags, opt influxql.IteratorOptions, cur integerCursor, aux []cursorAt, conds []*bufCursor, condNames []string) *integerIterator {
+	itr := &integerIterator{
+		cur: cur,
+		aux: aux,
+		opt: opt,
+		point: influxql.IntegerPoint{
+			Name: name,
+			Tags: tags,
+		},
+	}
+
+	if len(aux) > 0 {
+		itr.point.Aux = make([]interface{}, len(aux))
+	}
+
+	if opt.Condition != nil {
+		itr.m = make(map[string]interface{}, len(aux)+len(conds))
+	}
+	itr.conds.names = condNames
+	itr.conds.curs = conds
+
+	return itr
+}
+
+// Next returns the next point from the iterator.
+func (itr *integerIterator) Next() *influxql.IntegerPoint {
+	for {
+		seek := tsdb.EOF
+
+		if itr.cur != nil {
+			// Read from the main cursor if we have one.
+			itr.point.Time, itr.point.Value = itr.cur.nextInteger()
+			seek = itr.point.Time
+		} else {
+			// Otherwise find lowest aux timestamp.
+			for i := range itr.aux {
+				if k, _ := itr.aux[i].peek(); seek == tsdb.EOF || k < seek {
```

Comment:
There's a problem with this that I fixed in a recent change to the template for this file. Can you rebase and regenerate this file?